### DISCUSSION TITLE
How to copy images to system clipboard and not just the image path (SOLUTION)

### DISCUSSION THREAD

**Discussion by gordonn** [sxyazi/yazi#1525] [2. Show and tell]
If any of you want to copy and paste an image from Yazi to say a web browser and not just the directory path string then its actually quite simple:

Inside the keymap.toml add this keymap:

```
[manager]
prepend_keymap = [
    { on = "y", run = [
        '''
        shell 'xclip -selection clipboard -t image/png -i "$@"' --confirm
        ''',
        "yank"
    ], desc = "Yank image and copy it to clipboard" }
]
```

I started this discussion because I believe the docs didn't explicitly state how to do this (correct me if I am wrong). The closest thing I could find was this: https://yazi-rs.github.io/docs/tips/#selected-files-to-clipboard However this only copied the image path as a string and not the image itself.

**Comment by sxyazi** [comment]
Thanks for sharing! There's a plugin https://github.com/orhnk/system-clipboard.yazi you may also want to try

**Comment by rieje** [comment]
I'm curious if there are scenarios where you can't just [drag-drop](https://github.com/vimpostor/blobdrop) pictures from yazi for the same effect and leave `y` to copy filenames even if they are images. Simpler behavior and less mental overhead.

**Reply by minusfive** [reply]
Some of us (whether due to physical limitations or just preference), would rather keep our hands on the keyboard.

**Comment by gatzifratzi** [comment]
Can I replicate smth like this on mac?
